### PR TITLE
ariadne-codegen 0.14.0 (multiple outputs, test, add maintainer)

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

references:
- includes #1 

changes:
- [x] add `outputs`
- [x] run test suite (under coverage)
- [x] add maintainer

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by bollwyvl** [comment]
@conda-forge/ariadne-codegen ready for review

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,52 +1,125 @@
-{% set name = "ariadne-codegen" %}
-{% set version = "0.13.0" %}
+{% set version = "0.14.0" %}
+
+# handle undefined PYTHON in `noarch: generic` outputs
+{% if PYTHON is not defined %}{% set PYTHON = "$PYTHON" %}{% endif %}
```
Comment (ytausch) [inline recipe/meta.yaml]: I don't really understand why this is needed

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,52 +1,125 @@
...
-    - pip
+
+outputs:
+  - name: ariadne-codegen
+  - name: ariadne-codegen-with-websockets
+    build:
+      noarch: generic
+    requirements:
+      run:
+        - {{ pin_subpackage("ariadne-codegen", max_pin="x.x.x") }}
```
Comment (ytausch) [inline recipe/meta.yaml]: I think we want `exact=True` here instead, also below

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,52 +1,125 @@
...
+        - requests-toolbelt
+      imports:
+        - ariadne_codegen
+      commands:
+        - pip check
+        - ariadne-codegen --help
+        # doesn't handle windows slashes
+        {% set skips = '-k "not invalid_file"' %}  # [win]
+        {% set skips = '' %}                       # [unix]
+        - cd src && coverage run --source=ariadne_codegen --branch -m pytest -vv --tb=long --color=yes {{ skips }}
```
Comment (ytausch) [inline recipe/meta.yaml]: I am fine with running the tests for confirming correct packaging and detecting broken releases but I think we really don't need the coverage report or even fail if it's too low. That is up to the upstream maintainers.

**Review by ytausch — APPROVED** [review]
Thanks for contributing! I modified some details a bit and will merge right now for a quick release - if you want to discuss something further, let's do so!

File: recipe/meta.yaml (hunk 1 above)
Comment (ytausch) [reply recipe/meta.yaml]: Ah I think I got it. The build has to be moved from package-level to the first output for the `PYTHON` variable to be set.

Comment (ytausch) [reply recipe/meta.yaml]: ... and there needs a global Python host and run requirement, I took that from https://github.com/conda-forge/gql-feedstock/blob/main/recipe/meta.yaml

Comment (ytausch) [reply recipe/meta.yaml]: ... as well as in every subpackage

Comment (bollwyvl) [reply recipe/meta.yaml]: the `noarch: generic` saves the extra solves for each package, which is a better use of CI resources.

File: recipe/meta.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipe/meta.yaml]: `exact` matches down to the build hash... the hashes will never match between a `noarch: generic` and `noarch: python`

File: recipe/meta.yaml (hunk 3 above)
Comment (bollwyvl) [reply recipe/meta.yaml]: the coverage is an additional indicator that some new `[extra]` hasn't been packaged for which the feedstock doesn't account.

Comment (ytausch) [reply recipe/meta.yaml]: Understood, but I think we can get rid of the additional complexity by saying that additional `[extra]`s are never a breaking change, just additional features that we can support here

File: recipe/meta.yaml (hunk 1 above)
Comment (ytausch) [reply recipe/meta.yaml]: xref https://github.com/conda/conda-build/issues/5424

Comment (ytausch) [reply recipe/meta.yaml]: Agreed, see #3

File: recipe/meta.yaml (hunk 2 above)
Comment (ytausch) [reply recipe/meta.yaml]: Works in #3 though

**Pull request merged**